ui-tests/tests/widgets.test.ts

```diff
@@ -1,90 +1,47 @@
-    await galata.resetUI();
-    galata.context.capturePrefix = 'widgets';
-  });
+import { expect } from '@playwright/test';
 
-  afterAll(async () => {
-    galata.context.capturePrefix = '';
-  });
+import * as path from 'path';
 
-  test('Upload files to JupyterLab', async () => {
-    await galata.contents.moveDirectoryToServer(
-      path.resolve(__dirname, `./notebooks`),
-      'uploaded'
+test.describe('Widget Visual Regression', () => {
+  test.beforeEach(async ({ page, tmpPath }) => {
+    await page.contents.uploadDirectory(
+      path.resolve(__dirname, './notebooks'),
+      tmpPath
     );
-    expect(
-      await galata.contents.fileExists('uploaded/widgets.ipynb')
-    ).toBeTruthy();
-    expect(
-      await galata.contents.fileExists('uploaded/WidgetArch.png')
-    ).toBeTruthy();
-  });
-
-  test('Refresh File Browser', async () => {
-    await galata.filebrowser.refresh();
-  });
-
-  test('Open directory uploaded', async () => {
-    await galata.filebrowser.openDirectory('uploaded');
-    expect(
-      await galata.filebrowser.isFileListedInBrowser('widgets.ipynb')
-    ).toBeTruthy();
+    await page.filebrowser.openDirectory(tmpPath);
   });
 
-  test('Run notebook widgets.ipynb and capture cell outputs', async () => {
+  test('Run notebook widgets.ipynb and capture cell outputs', async ({
+    page,
+    tmpPath,
+  }) => {
     const notebook = 'widgets.ipynb';
-    await galata.notebook.open(notebook);
-    expect(await galata.notebook.isOpen(notebook)).toBeTruthy();
-    await galata.notebook.activate(notebook);
-    expect(await galata.notebook.isActive(notebook)).toBeTruthy();
+    await page.notebook.openByPath(`${tmpPath}/${notebook}`);
+    await page.notebook.activate(notebook);
 
-    let numCellImages = 0;
+    const captures = new Array<Buffer>();
+    const cellCount = await page.notebook.getCellCount();
 
-    const getCaptureImageName = (id: number): string => {
-      return `cell-${id}`;
-    };
-
-    await galata.notebook.runCellByCell({
+    await page.notebook.runCellByCell({
       onAfterCellRun: async (cellIndex: number) => {
-        const cell = await galata.notebook.getCellOutput(cellIndex);
+        const cell = await page.notebook.getCellOutput(cellIndex);
         if (cell) {
-          if (
-            await galata.capture.screenshot(
-              getCaptureImageName(numCellImages),
-              cell
-            )
-          ) {
-            numCellImages++;
-          }
+          captures.push(await cell.screenshot());
         }
       },
     });
 
-    for (let c = 0; c < numCellImages; ++c) {
-      expect(
-        await galata.capture.compareScreenshot(getCaptureImageName(c))
-      ).toBe('same');
-    }
-  });
-
-  test('Close notebook widgets.ipynb', async () => {
-    await galata.notebook.close(true);
-  });
-
-  test('Open home directory', async () => {
-    await galata.filebrowser.openHomeDirectory();
-  });
+    await page.notebook.runCellByCell();
```

Comment:
Are you executing the Notebook a second time here? Or did line 30 not execute the Notebook?

Reply:
Ah good catch, looks like we should be able to remove that line which is likely just a leftover from a previous iteration.

Reply:
```suggestion
```